qp_shotgun/humann2/humann2.py

```diff
@@ -11,23 +11,220 @@
 
 from future.utils import viewitems
 from functools import partial
+from subprocess import Popen
+from contextlib import contextmanager
+from tempfile import mkdtemp
+from shutil import rmtree
 
 from qiita_client import ArtifactInfo
 from qiita_client.util import system_call, get_sample_names_by_run_prefix
 
 
-def generate_humann2_analysis_commands(forward_seqs, reverse_seqs, map_file,
-                                       out_dir, parameters):
-    """Generates the HUMAnN2 commands
+@contextmanager
+def make_temp_directory():
```

Comment:
Probably ideally this will be local scratch if such a thing exists. How can we specify?